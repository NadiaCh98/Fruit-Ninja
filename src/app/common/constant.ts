export const SCENE_SIZE = 9;
