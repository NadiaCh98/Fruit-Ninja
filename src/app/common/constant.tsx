import { StartMenuItem } from '../game/models/startMenu';
import { GameConfig } from '../game/models/game';
import { ReactComponent as ExitIcon } from '../../assets/icons/close.svg';
import { ReactComponent as PauseIcon } from '../../assets/icons/pause.svg';
import { ReactComponent as ReplayIcon } from '../../assets/icons/replay.svg';
import { ReactComponent as PlayIcon } from '../../assets/icons/play.svg';

export const SCENE_SIZE = 9;
export const ATTEMPS = 3;
export const MIN_CUT_COMBO = 3;
export const DELAY_BETWEEN_FRUITS_GROUP = 3000;

export enum Fruits {
  apple,
  banana,
  lemon,
  pear
}

export enum FruitSpeed {
  Low = 2.8,
  Average = 3,
  Fast = 3.4,
}

export enum Game {
  Classic = 'Classic',
  Dzen = 'Dzen',
}

export const START_MENU: StartMenuItem[] = [
  {
    mode: Game.Classic,
    fruit: 'apple',
  },
  {
    mode: Game.Dzen,
    fruit: 'apple',
  },
];

export const GAME_CONFIG: GameConfig[] = [
  {
    game: Game.Classic,
    timer: null,
    attempts: 3,
  },
  {
    game: Game.Dzen,
    timer: 90000,
    attempts: 0,
  },
];

export enum PermissibleButton {
  Exit = 'Exit',
  Pause = 'Pause',
  Play = 'Play',
  Replay = 'Replay',
}

export const BUTTONS: Record<PermissibleButton, JSX.Element> = {
  Exit: <ExitIcon />,
  Pause: <PauseIcon />,
  Play: <PlayIcon />,
  Replay: <ReplayIcon />,
};

export const RIGHT_POSITION: React.CSSProperties = {
  top: 0,
  right: 0,
  width: 'max-content',
};
