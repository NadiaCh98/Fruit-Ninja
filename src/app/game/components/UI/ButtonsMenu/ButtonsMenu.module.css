.wrapper {
    padding: 2rem;
    background: linear-gradient(180deg, rgba(255,175,0,1) 0%, rgba(152,97,0,1) 100%);
    border: 3px solid #986100;
    border-radius: 7px;
    animation: wrapper 2s forwards;
}

@keyframes wrapper {
    from {
        transform: scale(0);
    }
    to {
        transform: scale(1);
    }
}

.menu {
    width: 100%;
    display: inline-flex;
    justify-content: center;
    gap: 20px;
}

.title {
    font-size: var(--text-size);
    color: white;
    text-align: center;
    margin: 0;
    margin-bottom: 0.5em;
}