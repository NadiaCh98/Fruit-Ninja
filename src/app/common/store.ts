import { GameStore } from '../gameStore';
import { FRUITS_POSITION_INTERVAL } from './constant';
import { createContext } from './storeUtils';

export const { StoreProvider, useStore } = createContext({
  Game: new GameStore(FRUITS_POSITION_INTERVAL),
});
