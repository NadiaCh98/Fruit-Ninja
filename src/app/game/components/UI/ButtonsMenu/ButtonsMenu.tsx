import React from 'react';
import { Button } from '../../../../common/components/Button/Button';
import { Modal } from '../../../../common/components/Modal/Modal';
import { PermissibleButton } from '../../../../common/constant';
import { ExitIcon, PauseIcon, PlayIcon, ReplayIcon } from '../Icons/Icons';
import styles from './ButtonsMenu.module.css';

const BUTTONS: Record<PermissibleButton, JSX.Element> = {
  Exit: <ExitIcon />,
  Pause: <PauseIcon />,
  Play: <PlayIcon />,
  Replay: <ReplayIcon />,
};

interface ButtonsMenuProps {
  readonly buttons: PermissibleButton[];
  readonly title: string;
  readonly buttonClick: (button: PermissibleButton) => void;
}

export const ButtonsMenu: React.FC<ButtonsMenuProps> = ({
  buttons,
  title,
  buttonClick,
}) => {
  const menuItemClick = React.useCallback(
    (button: PermissibleButton) => () => buttonClick(button),
    [buttonClick]
  );
  return (
    <Modal>
      <section className={styles.wrapper}>
        <h1 className={styles.title}>{title}</h1>
        <div data-testid="buttons" className={styles.menu}>
          {buttons.map((button, i) => (
            <Button key={i} clickHandler={menuItemClick(button)}>
              {BUTTONS[button]}
            </Button>
          ))}
        </div>
      </section>
    </Modal>
  );
};
