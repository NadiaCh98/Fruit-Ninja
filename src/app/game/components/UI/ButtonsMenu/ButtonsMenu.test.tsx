import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import { PermissibleButton } from '../../../../common/constant';
import { ButtonsMenu } from './ButtonsMenu';

describe('Buttons Menu', () => {
  const buttonClick = jest.fn();

  const buttons = [PermissibleButton.Exit];
  const buttonsMenu = (
    <ButtonsMenu
      buttonClick={buttonClick}
      buttons={buttons}
      title="Test Template"
    ></ButtonsMenu>
  );

  it('has a correct template', () => {
    const { container } = render(buttonsMenu);

    expect(container).toMatchSnapshot();
  });

  it('asserts buttons menu buttons correctly', () => {
    const { getByTestId } = render(buttonsMenu);
    expect(getByTestId('buttons').childElementCount).toBe(buttons.length);
  });

  it('assert buttonClick is fired correctly', () => {
    const { getByTestId } = render(buttonsMenu);
    fireEvent.click(getByTestId('buttons').childNodes[0]);
    expect(buttonClick).toBeCalledTimes(1);
    expect(buttonClick).toBeCalledWith(buttons[0]);
  });
});
