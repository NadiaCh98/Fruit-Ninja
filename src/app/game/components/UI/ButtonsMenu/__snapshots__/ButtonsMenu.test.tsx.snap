// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Menu has a correct template 1`] = `
<div>
  <div
    class="sc-gtsrHT dVXCLS"
  />
  <div
    class="sc-dlnjwi glLXEG"
  >
    <section
      class="wrapper"
    >
      <h1
        class="title"
      >
        Test Template
      </h1>
      <div
        class="menu"
        data-testid="buttons"
      >
        <button
          class="sc-bdnxRM dqJpqe"
          kind="default"
        >
          <svg>
            close.svg
          </svg>
        </button>
      </div>
    </section>
  </div>
</div>
`;
